Initial documentation review: getting-started (#303)
egil/bUnit: docs/site/docs/getting-started/fixture-details.md

**Setup and Test Methods:**

> Let us start by looking at the parameters that take a method as input first [__AP: do you mean that we'll first look at \<blah\> or that we'll look first at parameters that take a method as input?__]. The methods are called in the order they are listed below, and should be used for the described purpose:

Reviewer: ```suggestion
Let us start by looking at the parameters that take a method or lambda as input. The methods are called in the order they are listed below, and should be used for the described purpose:
```

_AP: do you mean that we'll first look at \<blah\> or that we'll look first at parameters that take a method as input?_

The latter. That we will look at the parameters that take parameters. Changed the sentence a bit, is that better?

Author: Yes fine now. With hindsight I was being really pernickety here!

 The `Test` or `TestAsync` method is called after the setup methods.   
  _One, and only one_ of the test methods can be specified per fixture. Use the test method to access the component under test and any fragments defined in the fixture, and interact and assert against them.
  
In the example above, the setup and test methods are declared in a `@code { }` block nested inside the <xref:Bunit.Fixture> component. This visually groups the methods nicely with the <xref:Bunit.Fixture> component, making it easier to see what parts of a test belong together, especially when you have multiple tests inside the same test component. 

You can place the methods anywhere you want inside the test component, which can be useful. For example, if you have the same setup steps for multiple tests, you can avoid code duplication by placing them in a common setup method that can be shared by the tests in the same test component.

> TODO EGIL [__AP: <== !!__]

Reviewer: ```suggestion
```

Lets just remove that TODO for now. I cannot figure out if it is already done or I simply cannot remember what ... todo.

Author: OK

For both `GetFragment` methods, the `id` string parameter is optional. If it is not provided, the first <xref:Bunit.Fragment> is used to return a <xref:Bunit.IRenderedFragment> or <xref:Bunit.IRenderedComponent`1>. Otherwise, the <xref:Bunit.Fragment> with an `Id` parameter that matches the `id` specified in the `GetFragment` method call will be used.

> The generic versions of <xref:Bunit.Fixture.GetComponentUnderTest``1> and <xref:Bunit.Fixture.GetFragment``1(System.String)> can specify a component of type `TComponent` which is not the first child of <xref:Bunit.ComponentUnderTest> or <xref:Bunit.Fragment>. This is useful in situations such as the component under test being wrapped inside a `<CascadingValue>`. The methods [__AP: method or methods?__] will return the _first_ component it [__AP: it or they?__] finds that matches the requested type through a depth-first search of the render tree.

Reviewer: ```suggestion
The generic versions of <xref:Bunit.Fixture.GetComponentUnderTest``1> and <xref:Bunit.Fixture.GetFragment``1(System.String)> can specify a component of type `TComponent` which is not the first child of <xref:Bunit.ComponentUnderTest> or <xref:Bunit.Fragment>. This is useful in situations such as the component under test being wrapped inside a `<CascadingValue>`. The `GetComponentUnderTest` and `GetFragment` methods will return the _first_ component they find that matches the requested type through a depth-first search of the render tree.
```

Does it make more sense now?

Author: Yes, perfect

## `<SimpleTodo>` Test Example

Let's look at an example of an elaborate test that tests the lifecycle of a simple task list component, `<SimpleTodo>` (listed below), which has a service injected, receives a cascading value, and changes between renders:

[!code-cshtml[SimpleTodo.razor](../../../samples/components/SimpleTodo.razor)]

In the test, we want to verify that:

- The `<form>` resets itself correctly after a task has been added
- The task is added correctly to the task list
- The "Theme" cascading value is correctly assigned to the task list

The test looks like this:

[!code-cshtml[SimpleTodoTest.razor](../../../samples/tests/razor/SimpleTodoTest.razor?highlight=4,5,8-10,13,20,29,30,35-37)]

Let's look at what's going on in this test:

1. The fixture has both a setup and test method specified. The setup method is used to register an empty list of tasks that the `<SimpleTodo>` component requires.
2. The `<SimpleTodo>` component is wrapped in a `<CascadingValue>` component which passes down the "Theme" cascading value.

> 3. The first `<Fragment>` does not have an `id`, since the `GetFragment()` method will pick the first fragment if no `id` is provided.
> 4. The second `<Fragment Id="expected tasks">` does have an `Id`, to make it possible to get the second fragment through a call to the `GetFragment("expected tasks")` method. [__AP: Id or id?__]

Reviewer: ```suggestion
3. The first `<Fragment>` does not have an `Id` parameter. It is not needed since the `GetFragment(string? id = null)` method will pick the first `<Fragment>` if no `id` argument is passed to it.
4. The second `<Fragment Id="expected tasks">` does have an `Id` parameter, to make it possible to get the second fragment through a call to the `GetFragment("expected tasks")` method.
```

Is this more clear now?

Author: It's fine, I think my question was more about the consistency of id vs. Id in terms of capitalization. The addition of 'parameter' makes me wonder it should be 'parameter' or 'attribute', since that's what it is in XML terms.
